### PR TITLE
FAI-873: Prepare for JBoss deployment

### PR DESCRIPTION
**Pull request by ruivieira**
Prepare POM for Nexus release.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,8 +7,33 @@
...
+        <url>scm:git:git@github.com:trustyai-explainability/trustyai-explainability.git</url>
+        <developerConnection>scm:git:git@github.com:trustyai-explainability/trustyai-explainability.git</developerConnection>
+    </scm>
+
+    <developers>
+        <developer>
+            <id>ruivieira</id>
+            <name>Rui Vieira</name>
+            <organization>Red Hat</organization>
+        </developer>
```
Comment (tteofili) [inline pom.xml:31]: we might include all of us, wdyt?

**Review by tteofili — APPROVED** [review]
LGTM, with a minor comment.

File: pom.xml [hunk 2]
```diff
@@ -7,8 +7,33 @@
     <artifactId>trustyaiexplainability</artifactId>
     <version>2.0.0-SNAPSHOT</version>
     <name>TrustyAI Explainability</name>
+    <description>TrustyAI is a java library for Explainable AI.</description>
+    <url>https://repository.jboss.org/nexus/content/repositories/trustyai-explainability/</url>
```
Comment (danielezonca) [inline pom.xml]: I think this is supposed to be TrustyAI website url

File: pom.xml [hunk 3]
```diff
@@ -7,8 +7,33 @@
...
     <packaging>pom</packaging>
 
+    <licenses>
+        <license>
+            <name>Apache License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+            <distribution>repo</distribution>
+            <comments>At least one license must be present. Here is an example of ASL v. 2.0</comments>
```
Comment (danielezonca) [inline pom.xml]: This is probably a placeholder, I think we should remove it

File: pom.xml (hunk 1 above)
Comment (ruivieira) [reply pom.xml:31]: Of course :) Added all developers

File: pom.xml [hunk 4]
```diff
@@ -5,10 +5,50 @@
...
+        <developer>
+            <id>RobGeada</id>
+            <name>Rob Geada</name>
+            <organization>Red Hat</organization>
+        </developer>
+        <developer>
+            <id>danielezonca</id>
+            <name>Daniele Zonca</name>
+            <organization>Red Hat</organization>
+        </developer>
```
Comment (danielezonca) [inline pom.xml]: We can probably also just link GH team page like
```suggestion
        <developer>
            <name>TrustyAI Explainability team</name>
            <url>https://github.com/orgs/trustyai-explainability/people</url>
        </developer>
```

Note: we need to make each of us `Public` in that page to show up

File: pom.xml [hunk 5]
```diff
@@ -5,10 +5,50 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.kie.trustyai</groupId>
     <artifactId>trustyaiexplainability</artifactId>
-    <version>2.0.0-SNAPSHOT</version>
+    <version>0.3.0</version>
```
Comment (danielezonca) [inline pom.xml]: Side note: we need to make sure the python version and java version are in sync

File: pom.xml (hunk 3 above)
Comment (ruivieira) [reply pom.xml]: IIUC the licence field is mandatory, but I've removed the `comment`.

Comment (danielezonca) [reply pom.xml]: Yes I was referring to the `comment` tag 👍

File: explainability-integrationtests/explainability-integrationtests-opennlp/pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.kie.trustyai</groupId>
     <artifactId>explainability-integrationtests</artifactId>
-    <version>2.0.0-SNAPSHOT</version>
+    <version>${revision}</version>
```
Comment (danielezonca) [inline explainability-integrationtests/explainability-integrationtests-opennlp/pom.xml]: I'm pretty sure this only works if I compile from the root of the project and not if I execute maven inside this module: it has a circular-dependency issue, `revision` is defined in the parent pom but it is also used here to "resolve" where/what is the parent.
Usually it is quite common to keep this fixed and use maven release plugin (or maven versions) to update it during the release

File: pom.xml [hunk 7]
```diff
@@ -197,6 +245,17 @@
                     <artifactId>maven-shade-plugin</artifactId>
                     <version>${version.shade.plugin}</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-release-plugin</artifactId>
+                    <version>3.0.0-M6</version>
+                    <configuration>
+                        <autoVersionSubmodules>true</autoVersionSubmodules>
+                        <developmentVersion>2.0.0-SNAPSHOT</developmentVersion>
```
Comment (danielezonca) [inline pom.xml]: Minor comment, this value is still related to Kogito, we can probably use a different policy (i.e. [quarkus](https://github.com/quarkusio/quarkus/blob/main/pom.xml#L16) uses `999-SNAPSHOT`)

**Review by danielezonca — APPROVED** [review]

File: pom.xml (hunk 7 above)
Comment (ruivieira) [reply pom.xml]: Good point. Changed to `999-SNAPSHOT` too.

**Pull request merged**